command/agent/fs_endpoint.go

```diff
@@ -355,48 +356,48 @@ func (s *HTTPServer) fsStreamImpl(resp http.ResponseWriter,
 	// Create a channel that decodes the results
 	errCh := make(chan HTTPCodedError)
 	go func() {
+		defer cancel()
+
 		// Send the request
 		if err := encoder.Encode(args); err != nil {
 			errCh <- CodedError(500, err.Error())
-			cancel()
 			return
 		}
 
 		for {
 			select {
 			case <-ctx.Done():
 				errCh <- nil
-				cancel()
 				return
 			default:
 			}
 
 			var res cstructs.StreamErrWrapper
 			if err := decoder.Decode(&res); err != nil {
 				errCh <- CodedError(500, err.Error())
-				cancel()
 				return
 			}
+			decoder.Reset(httpPipe)
```

Comment:
This doesn't look related to the race, was this all general cleanup?

Reply:
Yeah, and I was torn on whether to include it or not.

When testing adding or removing these lines had no impact on truncation or ordering issues.

However, they also didn't hurt and the docs say Reset should be called when "reusing" an encoder/decoder? https://godoc.org/github.com/ugorji/go/codec (search for Reset)

Reply:
Chatted with the team, and we're leaning toward keeping it Just In Case.